### DISCUSSION TITLE
Checking and related text

### DISCUSSION THREAD

**Discussion by Kerie70** [zippity21/Stardew_Valley_Regression_Mod#28] [Ideas]
So there is some flavor text that pops up from time to time when you have too big of an accident and ruin your pants, specifically your 'blue jeans'.  Same when you check your pants it's always blue jeans.  Would there be a way to implement a system that checks what type of pants you're wearing (between dress, skirt, shorts, pants) and have it relay text accordingly?  Given how colors can be all over the place with the RGB sliders could just drop the color part all together and just focus on the type of pants your character is wearing. 

Could also include unique text based on what you're wearing when checking yourself, not just your pants but your undies/pullups/diapers as well.  For dresses and skirts the text could involve lifting up your dress and skirt to check yourself.  A potential downside could be doing it around others you would essentially be flashing your underwear and the NPCs could react accordingly.

Could also add a randomizer system with other text for when you check yourself.  Right now it's a static response of pulling back the waistband and peering inside.  Other texts could be giving yourself a quick pat on the butt or front to see how squishy/used your diaper is.

**Comment by zippity21** [comment]
"Blue Jeans" was hardcoded in by the original author, probably as a temporary solution. Similarly for other static responses (they are **so much** easier).

Thankfully, the game seems to give us access to item names, and that includes what pants you're wearing.

What it doesn't give us is the color name. It does, however, give the color in a RGB format, so if I can figure out a course way of converting from RGB to text, that may be good enough; assuming we wanted to keep the color portion at all..


Unique text is difficult for two reasons:
1. Control Logic
The more factors that go into making a decision, the greater the likelihood of bugs, and the code can get ugly.

2. Writing the lines
I'm not particularly creative in that way, especially if you want a specific NPC have lines that others don't (or similar exception cases). I'd probably just need someone to hand me a set of lines, and I can write the logic around them.

That's not to say I can't or won't, because I like the idea. It just may take a while.
